### PR TITLE
Add event when saving drafts and make dw_locktimer reusable by plugin JS

### PR DESCRIPTION
**Pull request by micgro42**
Some plugins that completely replace the editor may still want to have their edit-progress be saved as drafts by the default DokuWiki draft-saving process.

- refactor the server-side draft saving to happen only at a single place and trigger a new event: `COMMON_DRAFT_SAVE`
- extend the the `dw_locktimer` to allow posting further fields to the server when saving drafts in the background while editing
- move `#draft__status` out of the toolbar `<div>` so we can hide the toolbar without hiding the draft status

### CODE REVIEW

File: inc/common.php [hunk 1]
```diff
@@ -1415,6 +1415,51 @@ function saveOldRevision($id) {
...
+    $event = new \Doku_Event('COMMON_DRAFT_SAVE', $draft);
+    if ($event->advise_before()) {
+        if (io_saveFile($draft['cname'], serialize($draft))) {
+            $draft['hasBeenSaved'] = true;
+            $INFO['draft'] = $draft['cname'];
+        }
+    }
+
+    $event->advise_after();
+    return (bool)$draft['hasBeenSaved'];
```
Comment (splitbrain) [inline inc/common.php]: `hasBeenSaved` will be uninitialized when an event prevented the default. It should probably be initialized as false further up.

**Review by splitbrain — APPROVED** [review]

File: lib/scripts/locktimer.js [hunk 2]
```diff
@@ -55,6 +58,15 @@ var dw_locktimer = {
         dw_locktimer.fieldsToSaveAsDraft.push(selector);
     },
 
+    /**
+     * Add a callback the gets executed the post to renew the lock / save the draft returns successfully
```
Comment (Klap-in) [inline lib/scripts/locktimer.js]: I do not get the meaning. Could you reformulate 'the gets executed'?

Comment (micgro42) [reply lib/scripts/locktimer.js]: Ah, this is a typo. `the` -> `that`

I will try to explain it in a bit more detail in the body of the doc block

File: lib/scripts/locktimer.js [hunk 3]
```diff
@@ -84,18 +111,21 @@ var dw_locktimer = {
...
         jQuery.post(
             DOKU_BASE + 'lib/exe/ajax.php',
             params,
-            dw_locktimer.refreshed,
+            null,
             'html'
-        );
+        ).done(function dwLocktimerRefreshDoneHandler(data) {
+            dw_locktimer.callbacks.forEach(
+                function foo(callback) {
```
Comment (splitbrain) [inline lib/scripts/locktimer.js]: Does this function have to be named? Wouldn't an anonymous function work?

**Comment by micgro42** [comment]
Closed in favor of #2415

**Pull request closed without merging**